### ISSUE TITLE
Wi-Fi Drivers Installation issue

### ISSUE THREAD

**Issue by francojoem** [Qengineering/Jetson-Nano-image#30]
I was using this Nano image for OpenCV+CUDA support and it was a huge relief, seeing all the packages pre-installed. I use the EdiMax EW-7811un Wi-Fi adapter. I tried installing the Wi-Fi drivers for the same, using the following guide : https://learn.sparkfun.com/tutorials/adding-wifi-to-the-nvidia-jetson/all , and understood that the Wi-Fi drivers weren't installing and connection to Wi-Fi was not possible still. But, it worked on the regular JetPack 4.6.3 version(the stock NVIDIA Image), without all the DNN frameworks. What could be the possible issue?

**Comment by Qengineering** [comment]
@francojoem,

I don't know. We don't have the EdiMax EW-7811un at the office, which makes testing a bit hard.
There are dozen different WiFi adapters. It is an impossible task to test and support them all.

The only thing I can come up with is a possible version difference. I've used an 'old' JetPack 4.6.1 as the base for the Nano image.
As you said, version 4.6.3 is the most actual. Perhaps a sudo apt update & upgrade?

The other way is starting with you working JetPack 4.6.3 and installing the packages yourself. The most time consuming will be OpenCV. The others are just wheels. Please follow [our guides](https://qengineering.eu/) on how to install these packages.

**Comment by francojoem** [comment]
Yeah, even had to do the same sudo apt-get update & upgrade, on the stock JetPack 4.6.3, to install the Wi-Fi Drivers for EdiMax. I  tried the same with the image of yours, and even after rebooting(after upgrading packages), it doesn't seem to work.

I actually converted to this Jetson image, because we were unsuccessful in building OpenCV with CUDA support for the JetPack 4.6.3(Always says low memory at the completion of the build, and just stops at 100%), following your guide. I was happy, that I was able to circumvent that using your Jetson image(which came with a lot of DNN packages installed), but the catch came in the form of not being able to install the Wi-Fi drivers for the same.

**Comment by Qengineering** [comment]
Did you enlarge the memory swap space before installing OpenCV?
OpenCV 4.7 needs a HUGE amount of memory to complete. Over 7 GByte.
And, of course, takes a lot of time to complete its last step. It can take hours!
Note also the free amount on your SD-card before maximizing your swap space.
It only enables the remaining free space on your card. Meaning that if you have only 4 GB free, your swap will be 4 GB.

**Comment by francojoem** [comment]
Had tried to install OpenCV 4.5.0 using the script provided in the Q-Engineering Website. Had created the swap file and changed the parameters according to the instructions in the Q-Engineering Website. 

I have a 128GB SD Card, so how do I create more than 4GB of swap memory, using the SD Card storage(By default, it is given for Swap Memory of 4GB, right)?

**Comment by Qengineering** [comment]
@francojoem,

Please look at the paragraph _Enlarge memory swap_ on the [guide](https://qengineering.eu/install-opencv-4.5-on-jetson-nano.html).
Here you see the installation of the dphys-swapfile software giving you as much swap memory as you wanted.
As said at the end, remove the package once done.

**Comment by francojoem** [comment]
What to do, when I need to get the Swap memory greater than 4026?

Is changing the CONF_SWAPSIZE to a value higher than 2048, is enough?

I tried tinkering with the CONF_MAXSWAP , but wasn't able to get swap memory higher than 4026(4025 to be exact), even after tinkering to a higher value than 4026.

**Comment by Qengineering** [comment]
You have two files.
`\sbin\dphys-swapfile` where you define the upper limit with CONF_MAXSWAP.
`\etc\dphys-swapfile` where you define the actual used memory with CONF_SWAPSIZE.
See how much space is left on your SD-card with the command `df -h`
And how large the current swapspace is with `free -m`
You can also check the reserved memory space by checking the file size of '\var\swap`
One thing for sure, 2048 will not be sufficient.

**Comment by francojoem** [comment]
Well, yeah I increased the variable to get a swap size of 11999, from the free -m command, and is currently installing OpenCV version 4.5.3, from the script provided on the Q-Engineering website. I would like to point that, updating the above mentioned points on the website could be highly helpful, since guys like me, who have just started tinkering with Linux distros, could understand, what variables to increase. As of now, there isn't much clarity to increase swap size beyond 4026, from the instructions of the website.

**Comment by Qengineering** [comment]
@francojoem,

I'm sorry to hear that the website isn't that clear. I've placed an extra line above the slide show, hoping readers will scroll though the slides.
![image](https://user-images.githubusercontent.com/44409029/235293329-10420dd8-b669-4a3b-8960-778b48098226.png)

**Comment by Beauxrel** [comment]
I was able to install the driver using the sparkfun instructions
make sure you install dkms with $sudo apt install dkms

**Comment by Qengineering** [comment]
@Beauxrel, Thanks !
